### ISSUE TITLE
WriteBufferManager: accurate memory tracking of immutable memory

### ISSUE THREAD

**Issue by isaac-io** [speedb-io/speedb#113]
Currently the WriteBufferManager tracks active memory through hooks into the memtable allocation and deallocation logic (using the `AllocTracker`. See [memtable.h](https://github.com/speedb-io/speedb/blob/1c7a33ae5ba2ce036696ba7d5baa4561bb59584a/db/memtable.h) and [allocator.h](https://github.com/speedb-io/speedb/blob/1c7a33ae5ba2ce036696ba7d5baa4561bb59584a/memory/allocator.h)).

This allows it to know how much mutable and immutable memory is currently taken up. However immutable memory isn't necessarily memory that can be released. For example, if a flush has been triggered on the immutable memory, that memory is already marked for deallocation, so triggering another flush there won't do any good.

For this reason, the WriteBufferManager currently does almost nothing with the immutable memory count. The current implementation takes action based on the immutable memory size only if capacity has been reached and half of that capacity is taken up by mutable memory.

We need to be able to know which immutable memory isn't already marked for flush, and for that the WriteBufferManager needs to hook into the flush logic in order to take that into account. Of note here is that memory is only marked for flush after picking memtables for flush (see [db_impl_compaction_flush.cc#L239](https://github.com/speedb-io/speedb/blob/0a7b4eb48581c8fb5a16103306b42b2aef800da5/db/db_impl/db_impl_compaction_flush.cc#L239) for regular flush and [db_impl_compaction_flush.cc#L503](https://github.com/speedb-io/speedb/blob/0a7b4eb48581c8fb5a16103306b42b2aef800da5/db/db_impl/db_impl_compaction_flush.cc#L503) for atomic flush), so there's still a span of time between triggering a flush and the time this memory will be marked as being in progress. An implementation note for when we start taking action based on immutable memory size in the future, is that we might need a way to wait for triggered flushed to pick memtables before we rely on the immutable memory size, but that's out of scope for this issue.

Note that we need to "unmark" the memory that is marked for flush when the flush is done and the immutable memory is released (which can be combined into the existing `FreeMem()` hook), as well as when the flush job is cancelled (such as when a CF is dropped) before memtables have been picked (in which case relying on `FreeMem()` would be the wrong thing to do because it would subtract memory marked for flush even though no memory has been marked yet.

**Comment by isaac-io** [comment]
Depends on #144.

**Comment by assaf-speedb** [comment]
Unit tests, stress and fuzz passed

**Comment by isaac-io** [comment]
This change adds two new functions to the WBM API:
* `immmutable_memtable_memory_usage()` - returns the amount of used memory that is taken up by immutable memory
* `memtable_memory_being_freed_usage()` - returns the amount of used memory that is currently being freed (this implies immutable memory).
